### ISSUE TITLE
Create a new project "Tour of Heroes app" Tutorial Produces Compile Errors 

### ISSUE THREAD

**Issue by kathleenwestpracticeC** [angular/angular-cli#20692]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

I don't know, this is my first time creating an Angular project following the tutorial instructions. 

 Yes, the previous version in which this bug was not present was: ....

### Description

 A clear and concise description of the problem...

I follow the "Tour of Heroes" Tutorial Project here
[https://angular.io/tutorial/toh-pt0](https://angular.io/tutorial/toh-pt0) and receive the build errors.

## 🔬 Minimal Reproduction

Do this in VS Code. Follow the Tour of Heroes Angular tutorial (https://angular.io/tutorial/toh-pt0) per the instructions to reproduce.

## 🔥 Exception or Error

<pre><code>

Error: ./src/main.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\......\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ multi ./src/main.ts main[0]

Error: ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\....\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ multi ./src/polyfills.ts polyfills[0]

Error: ./src/main.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\....\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ ./src/main.ts

Error: ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\.....\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ ./src/polyfills.ts

</code></pre>

## 🌍 Your Environment

<pre><code>

ng version 

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 12.0.0-rc.1
Node: 14.16.1
Package Manager: npm 6.14.12
OS: win32 x64

Angular: 12.0.0-rc.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0-next.8
@angular-devkit/build-angular   0.1200.0-next.8
@angular-devkit/core            12.0.0-next.8
@angular-devkit/schematics      12.0.0-rc.1
@schematics/angular             12.0.0-rc.1
rxjs                            6.6.7
typescript                      4.2.4

</code></pre>

**Anything else relevant?**

**Comment by kathleenwestpracticeC** [comment]
I did an accidental mouse click on the subject line and this ticket won't let me plugin in any edits or delete the original ticket to replace it with a more detailed one.

https://angular.io/tutorial/toh-pt0

Error: ./src/main.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\clearent\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ multi ./src/main.ts main[0]

Error: ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\clearent\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ multi ./src/polyfills.ts polyfills[0]

Error: ./src/main.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\clearent\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ ./src/main.ts

Error: ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/ivy/index.js):
TypeError: angularCompiler.getNextProgram is not a function
    at C:\clearent\angular-tour-of-heroes\node_modules\@ngtools\webpack\src\ivy\plugin.js:376:100
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
 @ ./src/polyfills.ts

**Comment by clydin** [comment]
Can you provide the output of `ng version` from within the project?

**Comment by kathleenwestpracticeC** [comment]
> Can you provide the output of `ng version` from within the project?

I just updated the ticket.

**Comment by clydin** [comment]
From the version output, it appears that multiple, different v12 prerelease versions are in use within the project.  Can you try updating the project `ng update @angular/cli --next`? The update should synchronize the versions.

**Comment by kathleenwestpracticeC** [comment]
> From the version output, it appears that multiple, different v12 prerelease versions are in use within the project. Can you try updating the project `ng update @angular/cli --next`? The update should synchronize the versions.

I did this and got 

`ng update @angular/cli --next
Using package manager: 'npm'
Collecting installed dependencies...
Found 26 dependencies.
Fetching dependency metadata from registry...
Package '@angular/cli' is already up to date.`

**Comment by kathleenwestpracticeC** [comment]
The build errors are still there after running ng update @angular/cli --next

**Comment by clydin** [comment]
Sorry about that.  The command should have been `ng update @angular-devkit/build-angular --next`.
Prerelease versions are unstable and can contain breaking changes. We recommend to update them as a group when new prereleases are available with the following command: `ng update @angular/cli @angular/core --next`.

**Comment by kathleenwestpracticeC** [comment]
Running this command 
`ng update @angular-devkit/build-angular --next` fixed things. The basic starter site was able to build and serve the site on my localhost. 

Can someone update the tutorial/starter website for this command?

**Comment by clydin** [comment]
The update command was particular to the state of the project referenced in this issue.
A newly generated project (`ng new ...`) should create a project with compatible versions of the framework packages.

Version 11 of Angular is the current, stable version of the framework and is recommended for production applications at this time. The command `npm install -g @angular/cli` should install the latest stable version.  Alternatively, to create a project without a global install or with particular version (v11 in this case), the following command can be used: `npx @angular/cli@11 new my-project`.

**Comment by kathleenwestpracticeC** [comment]
> The update command was particular to the state of the project referenced in this issue.
> A newly generated project (`ng new ...`) should create a project with compatible versions of the framework packages.
> 
> Version 11 of Angular is the current, stable version of the framework and is recommended for production applications at this time. The command `npm install -g @angular/cli` should install the latest stable version. Alternatively, to create a project without a global install or with particular version (v11 in this case), the following command can be used: `npx @angular/cli@11 new my-project`.

I followed the Angular website setup and tutorials with commands mentioned... so not sure why there were build errors on the new project.

I recommend someone update the Angular website/tutorials for those steps to make sure only the latest stable releases are being used and reach out to whoever is managing the releases to let them know that a simple new project does not build for:

@angular-devkit/build-angular   0.1200.0-next.8
@angular-devkit/core            12.0.0-next.8

**Comment by alan-agius4** [comment]
Hi @kathleenwestpracticeC, 

It's unclear to how you ended up using incompatible versions and moreover using version 12, which is currently the latest version. Following the steps of the tutorial provided https://angular.io/tutorial/toh-pt0, you should be on the latest version of Angular, which is currently version 11.

> I recommend someone update the Angular website/tutorials for those steps to make sure only the latest stable releases are being used and reach out to whoever is managing the releases to let them know that a simple new project does not build for:

>    @angular-devkit/build-angular 0.1200.0-next.8
>    @angular-devkit/core 12.0.0-next.8

> And I'd like to ask just how that code made it out into the wild when it cannot compile with a new project?

Again, it's unclear how you ended up in this state, we have several E2E testing the most used workflow, including generating a new project. I assure you that generating a new project using both latest and next version does work.

Can you please indicate the page(s) where it is referenced to use pre-release packages?

**Comment by kathleenwestpracticeC** [comment]
🤷‍♀️ This is the page [https://angular.io/guide/setup-local](url) 
I am a newbie to Angular so this is what I do, I follow the tutorials. When was the last time from Angular team followed the tutorials with a blank slate to see if it still works? 

This is my environment configuration now and the new project is working. 

`Angular CLI: 12.0.0-rc.1
Node: 14.16.1
Package Manager: npm 6.14.12
OS: win32 x64

Angular: 12.0.0-rc.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0-rc.1
@angular-devkit/build-angular   12.0.0-rc.1
@angular-devkit/core            12.0.0-rc.1
@angular-devkit/schematics      12.0.0-rc.1
@schematics/angular             12.0.0-rc.1
rxjs                            6.6.7
typescript                      4.2.4`

**Comment by alan-agius4** [comment]
Hi @kathleenwestpracticeC, 


Below are the commands listed in order of how they are mentioned in above page.
```
npm install -g @angular/cli
ng new my-app
cd my-app
ng serve --open
```

Following the steps in https://angular.io/guide/setup-local doesn't lead to a project being created with the prerelease version of the Angular mentioned above and surely not a broken project.

At this point, there isn't any actionable from our end, since the tutorial correct. Feel free to open a new issue with more information if you remember how you ended up having a project in a broken state.

Thanks.

**Comment by kathleenwestpracticeC** [comment]
@alan-agius4 
🤷‍♀️ This is the page https://angular.io/guide/setup-local
I am a newbie to Angular so this is what I do, I follow the tutorials and here we are. I assumed you verified it with a blank slate? 
"Saying it is so does not make it so"

**Comment by petebacondarwin** [comment]
@kathleenwestpracticeC - I wonder if you accidentally installed the `next` version of the CLI globally at some point?
Once that is done following the steps in that guide would indeed result in `next` version of the project.
Perhaps the first step in the guide should be changed to

```bash
npm install -g @angular/cli@latest
```

**Comment by kathleenwestpracticeC** [comment]
@petebacondarwin that sounds logical. I don't remember doing that, but it is possible, and if that were done, then that would explain it what happened. I did also install some Angular extensions in Node. If that command puts the environment at the latest stable release, then that is probably best to update the tutorial document to clarify that. I'm going to take note of that command for future reference.

**Comment by kathleenwestpracticeC** [comment]
🤦 Nothing changes when nothing changes

**Comment by kathleenwestpracticeC** [comment]
Well I am not logging into my personal google account on my work computer to sign code contributor agreements. Someone else feel free to handle this pull request/documentation update or do their own.
